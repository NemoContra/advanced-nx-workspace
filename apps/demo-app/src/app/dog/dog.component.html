<p>Only dogs here</p>
<flight-workspace-shared-util-random-animal-image></flight-workspace-shared-util-random-animal-image>
<br />
<button flightWorkspaceSharedUiAnimalSoundButton="dog">Make sound</button>
