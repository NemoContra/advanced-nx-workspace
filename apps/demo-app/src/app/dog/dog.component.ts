import { Component } from '@angular/core';

@Component({
  selector: 'flight-workspace-dog',
  templateUrl: './dog.component.html',
  styleUrls: ['./dog.component.css'],
})
export class DogComponent {}
