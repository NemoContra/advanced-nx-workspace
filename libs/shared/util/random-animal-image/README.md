# shared-util-random-animal-image

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-random-animal-image` to execute the unit tests.
