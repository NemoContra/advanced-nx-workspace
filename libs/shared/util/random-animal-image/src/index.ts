export * from './lib/shared-util-random-animal-image.module';
