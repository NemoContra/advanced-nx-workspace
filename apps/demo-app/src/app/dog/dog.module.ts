import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DogComponent } from './dog.component';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [DogComponent],
  imports: [
    CommonModule,
    SharedUtilRandomAnimalImageModule.forChild({ cats: false, dogs: true }),
    RouterModule.forChild([
      {
        path: '',
        component: DogComponent,
      },
    ]),
  ],
})
export class DogModule {}
