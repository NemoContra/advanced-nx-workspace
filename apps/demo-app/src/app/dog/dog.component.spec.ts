import { createRoutingFactory } from '@ngneat/spectator/jest';
import { DogComponent } from './dog.component';
import { MockModule } from 'ng-mocks';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';

describe('DogComponent', () => {
  const createComponent = createRoutingFactory({
    component: DogComponent,
    routes: [
      {
        path: '',
        component: DogComponent,
      },
    ],
    imports: [MockModule(SharedUtilRandomAnimalImageModule)],
  });

  it('should render', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
