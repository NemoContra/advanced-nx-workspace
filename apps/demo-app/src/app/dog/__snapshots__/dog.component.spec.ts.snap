// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogComponent should render 1`] = `
<flight-workspace-dog
  __ngContext__={[Function LRootView]}
>
  <p>
    Only dogs here
  </p><flight-workspace-shared-util-random-animal-image /><br /><button
    flightworkspaceshareduianimalsoundbutton="dog"
  >
    Make sound
  </button>
</flight-workspace-dog>
`;
